src/Microsoft.Extensions.OptionsModel/IOptionsWatcher.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.OptionsModel
+{
+    public interface IOptionsWatcher<out TOptions>
```

Comment:
Personally, I don't really mind the name change way. If everyone's happy I'm happy.